mono/mini/tramp-amd64.c

```diff
@@ -937,13 +937,25 @@ mono_arch_get_interp_to_native_trampoline (MonoTrampInfo **info)
 	guint8 *label_start_copy, *label_exit_copy;
 	MonoJumpInfo *ji = NULL;
 	GSList *unwind_ops = NULL;
-	int buf_len, i, off_methodargs, off_targetaddr;
+	int buf_len, i, cfa_offset, off_methodargs, off_targetaddr;
 
-	buf_len = 512 + MONO_TRAMPOLINE_UNWINDINFO_SIZE(0);
-	start = code = (guint8 *) mono_global_codeman_reserve (buf_len);
+	buf_len = 512;
+	start = code = (guint8 *) mono_global_codeman_reserve (buf_len + MONO_MAX_TRAMPOLINE_UNWINDINFO_SIZE);
+
+	// CFA = sp + 8
+	cfa_offset = 8;
+	mono_add_unwind_op_def_cfa (unwind_ops, code, start, AMD64_RSP, cfa_offset);
+	// IP saved at CFA - 8
+	mono_add_unwind_op_offset (unwind_ops, code, start, AMD64_RIP, -cfa_offset);
 
 	amd64_push_reg (code, AMD64_RBP);
+	cfa_offset += sizeof (target_mgreg_t);
+	mono_add_unwind_op_def_cfa_offset (unwind_ops, code, start, cfa_offset);
+	mono_add_unwind_op_offset (unwind_ops, code, start, AMD64_RBP, -cfa_offset);
+
 	amd64_mov_reg_reg (code, AMD64_RBP, AMD64_RSP, sizeof (target_mgreg_t));
+	mono_add_unwind_op_def_cfa_reg (unwind_ops, code, start, AMD64_RBP);
+	mono_add_unwind_op_fp_alloc (unwind_ops, code, start, AMD64_RBP, 0);
```

Comment:
do you know what this unwind up actually does ?

Reply:
Sets the frame pointer register + inform unwinder about offset. This is mainly used by Win x64 unwind op codes since this information is part of the Win x64 ABI.

Reply:
The fp_alloc is a noop on none Windows platforms.